test/framework/src/main/java/org/elasticsearch/cluster/MockInternalClusterInfoService.java

```diff
@@ -94,27 +93,35 @@ List<NodeStats> adjustNodesStats(List<NodeStats> nodesStats) {
         }).collect(Collectors.toList());
     }
 
-    class SizeFakingClusterInfo extends ClusterInfo {
-        SizeFakingClusterInfo(ClusterInfo delegate) {
-            super(
-                delegate.getNodeLeastAvailableDiskUsages(),
-                delegate.getNodeMostAvailableDiskUsages(),
-                delegate.shardSizes,
-                delegate.shardDataSetSizes,
-                delegate.routingToDataPath,
-                delegate.reservedSpace
-            );
+    @Override
+    ShardStats[] adjustShardStats(ShardStats[] shardsStats) {
+        var shardSizeFunctionCopy = this.shardSizeFunction;
```

Comment:
I think at one point we did need this to be lazier, but not any more it seems. In fact today all implementations are just constant functions returning either 0 or 1 so there's not really a need for this to be a function at all.